compiler/modules/front.syntax/tests/testParserOnTestSuite.bal

```diff
@@ -44,3 +63,85 @@ function partToLines(ModulePart part) returns string[] {
 function scanAndParseModulePart(string[] lines, FilePath path, int partIndex) returns ModulePart|err:Syntax {
     return parseModulePart(check scanModulePart(createSourceFile(lines, path), partIndex));
 }
+
+function testIsWhitespace(SourceFile file, Position startPos, Position endPos) returns boolean {
+    err:LineColumn st = file.lineColumn(startPos);
+    err:LineColumn end = file.lineColumn(endPos);
+    int startLineIndex = st[0];
+    int endLineIndex = end[0];
+    int startFragIndex = fragIndex(st, file.line(startLineIndex));
+    int endFragIndex = fragIndex(end, file.line(endLineIndex));
+    int lineIndex = startLineIndex;
+    int i = st[1];
+    while lineIndex <= endLineIndex {
+        ScannedLine line = file.line(lineIndex);
+        while i < line.fragments.length() {
+            if lineIndex >= endLineIndex && i >= endFragIndex {
+                return true;
+            }
+            FragCode frag = line.fragCodes[i];
+            if frag != FRAG_WHITESPACE && frag != FRAG_COMMENT {
+                return false;
+            }
+            i += 1;
+        }
+        i = 0;
+        lineIndex += 1;
+    }
+    return true; // start == end
+}
+
+function fragIndex(err:LineColumn lineColumn, ScannedLine line) returns int {
```

Comment:
I would hope to have a method on sourceFile that takes a Position and returns the FragCode at that position. This is a generally useful function that would not be just in test.

Reply:
Moved the function to `SourceFile`